code/src/components/QuestLog.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+
+import { Quest } from 'components/Quest'
+
+export const QuestLog = () => {
+  const questsList = useSelector((state) => state.quests.questsList)
+
+  const mainQuests = []
+  const sideQuests = []
+  const miscellaneousQuests = []
+
+  questsList.forEach((e, index) => {
+    if (e.category === 'Main') {
+      mainQuests.push(index)
+    } else if (e.category === 'Side') {
+      sideQuests.push(index)
+    } else {
+      miscellaneousQuests.push(index)
+    }
+  })
+
+  return (
+    <main>
+      {(mainQuests.length > 0)
+        && <>
+          <h3>Main</h3>
+          {mainQuests.map((e) => (
+            <Quest key={`quest-${e}`} questIndex={e} />))}
```

Comment:
When you have the entire quest object, why only pass the index? I think the `Quest` component could've just used the quest you pass it rather than looking into the store itself.